feat(php): add custom client section in readme

stash-rl opened the pull request:
## Description
Add documentation for using a custom client in the README. This makes it easier for our customers to visualize how they can override the default client.

## Changes Made
- Generate documentation describing how to use a custom client and include it in the README.

## Testing
<!-- Describe how you tested these changes -->
- [ ] Unit tests added/updated
- [x] Manual testing completed

stash-rl committed:
Add documentation for using a custom client in the README, this makes it easier for our customers to visualize how they can override the default client.

stash-rl committed:
Update generators/php/sdk/features.yml

Co-authored-by: graphite-app[bot] <96075541+graphite-app[bot]@users.noreply.github.com>

dsinghvi commented on seed/php-sdk/api-wide-base-path/README.md:
```diff
@@ -56,6 +56,19 @@ try {
 
 ## Advanced
 
+### Custom Client
```
https://github.com/square/square-php-sdk?tab=readme-ov-file#custom-http-client @stash-rl can you make the working closer to this

stash-rl replied:
Yepp sure 👍

Swimburger committed:
fix(internal): Make ts-sdk seed.yml consistent with csharp (#7790)

* Make seed.yml consistent with csharp

* fix

Commit:
chore(typescript): release multi-platform builds of the docker container

davidkonigsberg committed:
feat(ci): Validate versions.yml files (#7793)

* add versions validation ci

* run as matrix

* Update lint-versions-files.yml

* remove intentional breakage from testing

* Run on all pull requests since it will be required

stash-rl committed:
Clean up the generated readme.

sahil485 committed:
chore(cli): upgrade fdr and fdr-cjs deps to 0.138.3 (#7792)

stash-rl committed:
Fix a small typo.

dsinghvi reviewed (approved):
👏 looks great -- if you want to add something below the existing section about adding middleware to guzzle that would make sense to me as well but not required

Swimburger committed:
feat(typescript): Release 2.0.0 of the TypeScript SDK generator 🎉  (#7763)

* Release 2.0

---------

Co-authored-by: Deep Singhvi <deep@buildwithfern.com>

davidkonigsberg committed:
fix(ci): Improve versions.yml validation CI job names (#7794)

* rename and cleanup versions.yml validation

* hardcode job names

* Put all validations in a single job

* Revert "Put all validations in a single job"

This reverts commit 9e39872562b54b93b553117d54c69561addf7065.

* Make name more concise

* Update validate-versions-files.yml

* Update validate-versions-files.yml

* Update validate-versions-files.yml

stash-rl committed:
Add middleware usage example

stash-rl commented on generators/php/sdk/src/readme/ReadmeSnippetBuilder.ts:
```diff
@@ -121,6 +122,33 @@ $response = ${this.getMethodCall(timeoutEndpoint)}(
+]);
+
+// Pass the custom client when creating an instance of the class.
+${this.context.getClientVariableName()} = new ${this.context.getRootClientClassName()}(options: [
+    '${this.context.getGuzzleClientOptionName()}' => $customClient
+]);
```
I don't think is valid btw.

dsinghvi replied:
@stash-rl can you make this valid in the generated code snippet?

davidkonigsberg committed:
fix(changelog): Changelogs had not been updated, this updates them (#7797)

update changelogs

Swimburger committed:
fix(typescript): Change 2.0.0 release date to 2025-07-02 in versions.yml (#7799)

Change 2.0.0 release date to 2025-07-02 in versions.yml

fern-support committed:
chore(typescript): update typescript-sdk seed (#7800)

Co-authored-by: Swimburger <Swimburger@users.noreply.github.com>

chdeskur committed:
chore(docs): restore api ref (#7801)

Co-authored-by: fern-bot <info@buildwithfern.com>

Ryan-Amirthan committed:
fix(docs): Update preview-your-sdk-locally.mdx (#7803)

Update preview-your-sdk-locally.mdx

davidkonigsberg committed:
chore(ci): Consolidate versions.yml validation (#7802)

* Attempt to use shared action

* Update validate-versions-files.yml

* Update validate-versions-files.yml

fern-support committed:
chore(changelog): update changelog (#7804)

chore: update changelog

Co-authored-by: davidkonigsberg <72822263+davidkonigsberg@users.noreply.github.com>

davidkonigsberg committed:
fix(ci): When changelog updating CI was modified to run on main, I forgot to add a branch name for commits (#7798)

* Update write-changelogs.yml

sahil485 committed:
feat(cli): handle well known types and prepend all type names/references with package for namespacing (#7805)

* feat(grpc): handle well known types and prepend all type names/references with package for namespacing

* remove version pin

* formatting

fern-support committed:
chore(changelog): update changelog (#7806)

Co-authored-by: sahil485 <46851548+sahil485@users.noreply.github.com>

Swimburger committed:
feat(typescript): Split up Jest configuration into multiple projects (#7767)

* Improve jest configs

fern-support committed:
chore(changelog): update changelog (#7807)

Co-authored-by: Swimburger <3382717+Swimburger@users.noreply.github.com>

fern-support committed:
chore(typescript): update typescript-sdk seed (#7808)

Co-authored-by: Swimburger <Swimburger@users.noreply.github.com>

sahil485 committed:
fix(cli): don't use default documentation from well known types, add docs for fields (#7809)

* fix(cli): don't use default documentation from well known types, add docs for fields

* add versions.yml

* use literal for source code path

* format

fern-support committed:
chore(changelog): update changelog (#7811)

Co-authored-by: sahil485 <46851548+sahil485@users.noreply.github.com>

Swimburger committed:
chore(typescript): Remove browser-specific TS generator code (#7814)

* Remove browser-specific TS generator code

Swimburger committed:
chore(cli): Remove deprecated Redocly warning (#7816)

Remove deprecated Redocly warning

Swimburger committed:
feat(typescript): Improve generated package.json files (#7815)

Improve generated package.json files

fern-support committed:
chore(changelog): update changelog (#7817)

Co-authored-by: Swimburger <3382717+Swimburger@users.noreply.github.com>

Swimburger committed:
fix(typescript): Ensure tests/wire is created (#7818)

* Ensure tests/wire is created

fern-support committed:
fix(lint): auto-fix lint issues (#7819)

Co-authored-by: Swimburger <Swimburger@users.noreply.github.com>

fern-support committed:
chore(changelog): update changelog (#7820)

Co-authored-by: Swimburger <3382717+Swimburger@users.noreply.github.com>

fern-support committed:
chore(changelog): update changelog (#7821)

Swimburger committed:
feat(typescript): Change the outputSourceFiles default from false to true (#7812)

* Change the outputSourceFiles default from false to true

fern-support committed:
chore(changelog): update changelog (#7822)

Co-authored-by: Swimburger <3382717+Swimburger@users.noreply.github.com>

Commit:
fix(typescript): instantiate oauth client with global params (#7823)

Commit:
fix(typescript): update latest version to `2.3.1`

fern-support committed:
chore(changelog): update changelog (#7824)

Co-authored-by: dsinghvi <10870189+dsinghvi@users.noreply.github.com>

fern-support committed:
chore(typescript): update typescript-sdk seed (#7826)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

kafkas committed:
chore(swift): scaffold swift generator and wire up seed (#7829)

* Init `base` pkg

* Format

* Init `model` pkg

* Init `sdk` pkg

* Update lockfile

* Init seed/swift-sdk

* Snippet and snippet templates

* Add dockerfiles

* Add autogenerated files

Swimburger committed:
fix(typescript): Remove ".js" extension from ESM imports in the source generator code (#7828)

Remove ".js" extension from ESM imports in the source generator code.

fern-support committed:
chore(changelog): update changelog (#7832)

Co-authored-by: Swimburger <3382717+Swimburger@users.noreply.github.com>

sahil485 committed:
feat(cli): add proto method type to HttpEndpoint source IR (#7833)

* feat(cli): add proto method type to HttpEndpoint source IR

* make methodType optional

Swimburger committed:
chore(typescript): Rename fernapi/fern-typescript-node-sdk to fernapi/fern-typescript-sdk (#7825)

Commit:
chore(ci): setup separate depcheck job and fix (#7838)

wodow committed:
feat(python): Support variables in Python SDK generator (#7827)

sahil485 committed:
fix(cli): update fern check CI snapshots, update fdr dependencies for grpc methodType (#7855)

* fix(cli): update fern check CI snapshots, update fdr dependencies for grpc methodType

* update snapshots

Commit:
chore(ci): fix ete tests (#7858)

fern-support committed:
chore(openapi): update openapi seed (#7840)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(postman): update postman seed (#7839)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(go): update go-model seed (#7844)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(go): update go-fiber seed (#7843)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(java): update java-spring seed (#7842)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(python): update pydantic seed (#7848)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(php): update php-model seed (#7845)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

Commit:
chore(ci): fix eslint and make a required check (#7859)

fern-support committed:
fix(lint): auto-fix lint issues (#7860)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

Swimburger committed:
chore(cli): Rename the TypeScript SDK generator image from fernapi/fern-typescript-node-sdk to fernapi/fern-typescript-sdk (#7861)

Rename the TypeScript SDK generator image from fernapi/fern-typescript-node-sdk to fernapi/fern-typescript-sdk

fern-support committed:
chore(changelog): update changelog (#7862)

fern-support committed:
chore(java): update java-sdk seed (#7863)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(python): update python-sdk seed (#7864)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

fern-support committed:
chore(changelog): update changelog (#7865)

fern-support committed:
chore(typescript): update typescript-sdk seed (#7866)

Co-authored-by: dsinghvi <dsinghvi@users.noreply.github.com>

kafkas committed:
chore(ci): delete `writeDotMock` from seed (#7831)

Commit:
fix

stash-rl committed:
Add a clearer message
